classes/forbidden_shop/ForbiddenShopManager.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+require_once __DIR__ . '/../event/LanternEvent.php';
+
+class ForbiddenShopManager {
+    private System $system;
+    private User $player;
+
+    public function __construct(System $system, User $player) {
+        $this->system = $system;
+        $this->player = $player;
+    }
+
+    /**
+     * @param string $event_name
+     * @param string $curreny_name
+     * @param int $quantity
+     * @throws RuntimeException
+     */
+    public function exchangeEventCurrency($event_name, $currency_name, $quantity): string
+    {
+        $this->player->getInventory();
+        switch ($event_name) {
+            case "festival_of_shadows":
+                $yen_per_lantern = LanternEvent::$static_config['yen_per_lantern'];
+                $red_lantern_id = LanternEvent::$static_item_ids['red_lantern_id'];
+                $blue_lantern_id = LanternEvent::$static_item_ids['blue_lantern_id'];
+                $violet_lantern_id = LanternEvent::$static_item_ids['violet_lantern_id'];
+                $gold_lantern_id = LanternEvent::$static_item_ids['gold_lantern_id'];
+                $shadow_essence_id = LanternEvent::$static_item_ids['shadow_essence_id'];
+                switch ($currency_name) {
+                    case "all":
```

Comment:
Switch statement with one case doesn't make a lot of sense to me, we could rewrite as `if($currency_name == "all")` but what cases would that not be true in?